### ISSUE TITLE
AVM2: HitTestObject with SimpleButton not working

### ISSUE THREAD

**Issue by CUB3D** [ruffle-rs/ruffle#8722]
### Describe the bug

As shown by the attached sample, trying to perform a `hitTestObject`, where one value is a SimpleButton always fails.

Adding some additional logging around the current hit test logic, shows that the bounds of the button are always 
`BoundingBox { x_min: Twips(0), y_min: Twips(0), x_max: Twips(0), y_max: Twips(0), valid: false }`

This causes an issue in [Upgrade Complete!](https://www.newgrounds.com/portal/view/499812), where neither the Play button, nor the Back button in the Ship Builder work. This also might be the cause of the issue that prevents collecting money.

(In this SWF, click to move the green clip)
[UGC Bug.zip](https://github.com/ruffle-rs/ruffle/files/10182037/UGC.Bug.zip)

### Expected behavior

hitTestObject behavior matches Flash Player

### Affected platform

Desktop app

### Operating system

Linux

**Comment by CUB3D** [comment]
After playing about with this for a while, I think I've found *a* solution to this issue, currently the bounds of SimpleButton are taken as empty [here](https://github.com/ruffle-rs/ruffle/blob/2bb4e2e549ab39d2f2cb7d8194ac21bb1ca2e682/core/src/display_object/avm2_button.rs#L643).

Changing this to 
```
BoundingBox {
    x_max: Twips::ZERO,
    y_max: Twips::ZERO,
    x_min: Twips::ZERO,
    y_min: Twips::ZERO,
    valid: true,
}
```
Makes the button clickable (somewhat, the mouse seems offset but I doubt that is related). This also fixes the issue in the linked swf.

**Comment by n0samu** [comment]
Fixed by #8744
